### ISSUE TITLE
Build instructions don't work

### ISSUE THREAD

**Issue by dietmarkuehl** [beman-project/Optional26#31]
The build instructions of the `README.md` don't work. At least, it is required to get the submodules (as is done by the Makefile):

    git submodule update --init --recursive

Also, for the tests I get a message like (I'm building on a Raspberry PI with manually build gcc-14):

    Test project /home/kuehl/src/Optional26/.build/gcc-14
          Start  1: OptionalTest.TestGTest
    Test not available without configuration.  (Missing "-C <config>"?)
    1/82 Test  #1: OptionalTest.TestGTest ...........................***Not Run   0.00 sec

**Comment by neatudarius** [comment]
@dietmarkuehl , please check latest main after #29 .

**Comment by neatudarius** [comment]
Feedback gathered from Dietmar: `I still needed to dig about with files in the repo.`

Status: validated on RPI Linux with gcc-13 and clang-18. Failed for gcc-12. Need to lower cmake version to `3.25.1` and to remove generator from `CMakePresets.json`.  I would be nice to have `./configure; make; make install` or `cmake --build dir` shorthands.

- [ ] Test and document on non-Linux systems (e.g., MAC OS, Windows).
- [ ] Test and document minimum cmake and compiler versions.
- [ ] Test and document symlink to compilers (where should they be installed - e.g. clang++ vs clang++-18).
    - [ ] Symlinks for custom compilers.
    - [ ] Symlink for `c++` tool. 
- [ ] Test and document dependencies. `ninja, make or another Cmake-supported build system` currently `ninja` is enforced, but not clearly documented. Note: ninja is used only for generator in the preset!
- [ ] Test and document C++ standard versions.
- [ ] Test and document shorthands for `cmake --build dir` or make equivalent.

**Comment by camio** [comment]
> I would be nice to have `./configure; make; make install`

I don't think the additional complexity of supporting that is worth it. CMake alone is fine as a build system for C++ projects IMO.

**Comment by neatudarius** [comment]
> I don't think the additional complexity of supporting that is worth it. CMake alone is fine as a build system for C++ projects IMO.

@camio , agree with you. I think latest main README.md made progress in simplifing the build proccess of the project.  We can put more defaults params in the CMakeLists.txt and maybe we can even simplify more.  But we need to know if we want to invest or not time in this direction.

@camio , please also check https://discourse.bemanproject.org/t/possible-high-priority-topic-short-long-term-approach-for-build-interface/153 (linked issue there) - I think we should keep #31 to track docs for what we have on main as a build system and the discourse for eventually changing an approach.

**Comment by dietmarkuehl** [comment]
> > I would be nice to have `./configure; make; make install`
> 
> 
> 
> I don't think the additional complexity of supporting that is worth it. CMake alone is fine as a build system for C++ projects IMO.

I'm not disagreeing, assuming the cmake works. It may still be cute to effectively pretend (although I'd try to forward various arguments like --prefix to cmake):

$ cat configure
#!/bin/sh
echo configured

$ cat Makefile
.PHONY: default install
default:
    cmake <documented-defaults>

install:
    cmake <install-parameters>

**Comment by neatudarius** [comment]
I will close this issue as was fixed. Build instructions from main are more precise. Feel free to open other issue if you find other bugs.
